components/HomepageSlider/HomepageSlider.tsx

```diff
@@ -0,0 +1,105 @@
+import { useState, useEffect } from 'react';
+import { useOnWindowResize } from '../../custom_hooks/useOnWindowResize';
+import styles from "./HomepageSlider.module.scss";
+import Slide from './Slide';
+
+export default function HomepageSlider() {
+    const slide_amount = 2 // starts at 0
+    const [slide_size, setSlideSize] = useState(0);
+    const [slide_distance, setSlideDistance] = useState(0);
+    const [slide_position, setSlidePosition] = useState(0);
+    const [slide_card, setSlideCard] = useState(false);
+    const [mouse_down_start, setMouseDownStart] = useState(0);
+    const window_resize = useOnWindowResize();
+
+    const nextOrPrevSlide = (direction:string) => {
+        if (direction === "previous" && !(slide_position <= 0)) {
+            setSlidePosition(slide_position - 1);
+        }
+        if (direction === "next" && !(slide_position >= slide_amount)) {
+            setSlidePosition(slide_position + 1);
+        }
+    }
+
+    const slideCard = (mouse_x:number) => {
+        const mouse_moves_right = (mouse_x >= (mouse_down_start + 10));
+        const mouse_moves_left = (mouse_x <= (mouse_down_start - 10));
+        const slide_right = slide_card && mouse_moves_right && !(slide_position <= 0);
+        const slide_left = slide_card && mouse_moves_left && !(slide_position >= slide_amount);
+        if (slide_right) {
+            setSlidePosition(slide_position - 1);
+            setSlideCard(false);
+        } else if (slide_left) {
+            setSlidePosition(slide_position + 1);
+            setSlideCard(false);
+        } else {
+            return;
+        }
+    }
+
+    useEffect(() => {
+        setSlideDistance(slide_size * slide_position);
+    }, [slide_size, slide_position])
+
+    useEffect(() => {
+        if (window_resize && slide_distance !== 0) {
+            setSlideDistance(0);
+            setSlidePosition(0);
+        }
+    }, [window_resize])
```

Comment:
this will throw warning in command line since setSlideDistance and setSlidePosition is not in the dependency arreay

Reply:
Hello Ali, I believe in the dependency array of an useEffect only `states` are necessary to watch if one of those states is changing -> then do something based on that. I believe adding the `state setters` are not necessary. Looks like it is also mentioned inside the react documentation: https://reactjs.org/docs/hooks-reference.html#usestate

Reply:
I think i couldn't completely describe my point, I was talking about [this](https://stackoverflow.com/a/60327893/8699388)